src/components/Widgets/Markdown/MarkdownControl/RawEditor/index.js

```diff
@@ -22,16 +22,14 @@ export default class RawEditor extends React.Component {
     this.setState({ editorState });
   }
 
-  onChange = debounce(this.props.onChange, 250);
-
   /**
    * When the document value changes, serialize from Slate's AST back to plain
    * text (which is Markdown) and pass that up as the new value.
    */
-  handleDocumentChange = (doc, editorState) => {
+  handleDocumentChange = debounce((doc, editorState) => {
```

Comment:
I haven't done that much on the editor stuff, but the debounce fix looks great! Thanks!

Reply:
Yeah it should help a lot with all of the heavy Immutable/React plumbing behind Slate. Night and day from the current editor on large docs.